File: 이수창/20230514/올바른 괄호.js
function solution(s){
    let p = 0;
    let l = s.length;

    for (let a=0; a<l; a++) {
        if (s[a] === "(") {
            p = p + 1;
        } else {
            if (p < 1) return false;
            p = p - 1;
        }
    }

    return p > 0 ? false : true;
}

//괄호==스택 국룰
// 연산속도를 위해 배열 대신 더하기 뺴기 연산을 이용했다

File: 장주성/20230514/올바른 괄호.py
def solution(s):
    if s[0] == ')':
        return False
    # ')'로 시작하면 False

    opening_count = 0
    # 여는 괄호 개수 초기화

    for i in s:
        if i == '(':
            opening_count += 1
        # 여는 괄호 추가될 때마다 카운트 +1
        else:
            opening_count -= 1
        # 닫는 괄호는 카운트 -1
        if opening_count < 0:
            return False
        # 여는 괄호의 개수가 음수이면 False

    if opening_count > 0:
        return False
    # 전 과정이 끝난 후에 괄호의 개수가 양수이면 False

    return True
